Packaging Failure

Currently get build failures when trying to package a project with FaceFX enabled:

"MainFrameActions: Packaging (Windows (64-bit)): UnrealBuildTool: ERROR: Couldn't find module rules file for module 'FaceFX'."

Output log:

MainFrameActions: Packaging (Windows (64-bit)): Project.Build: ********** BUILD COMMAND STARTED **********
MainFrameActions: Packaging (Windows (64-bit)): InternalUtils.SafeFileExists: SafeFileExists C:\Program Files (x86)\Xoreax\IncrediBuild\xgConsole.exe=False
MainFrameActions: Packaging (Windows (64-bit)): UE4Build.Build: XGE was requested, but is unavailable, so we won't use it.
MainFrameActions: Packaging (Windows (64-bit)): UE4Build.Build: ************************* UE4Build:
MainFrameActions: Packaging (Windows (64-bit)): UE4Build.Build: ************************* ForceMonolithic: False
MainFrameActions: Packaging (Windows (64-bit)): UE4Build.Build: ************************* ForceNonUnity:False 
MainFrameActions: Packaging (Windows (64-bit)): UE4Build.Build: ************************* ForceDebugInfo: False
MainFrameActions: Packaging (Windows (64-bit)): UE4Build.Build: ************************* UseXGE: False
MainFrameActions: Packaging (Windows (64-bit)): InternalUtils.SafeFileExists: SafeFileExists D:\Program Files\Epic Games\4.8\Engine\Binaries\DotNET\UnrealBuildTool.exe=True
MainFrameActions: Packaging (Windows (64-bit)): InternalUtils.SafeFileExists: SafeFileExists D:\Program Files\Epic Games\4.8\Engine\Binaries\DotNET\UnrealBuildTool.exe=True
MainFrameActions: Packaging (Windows (64-bit)): InternalUtils.SafeFileExists: SafeFileExists D:\Program Files\Epic Games\4.8\Engine\Binaries\DotNET\UnrealBuildTool.exe=True
MainFrameActions: Packaging (Windows (64-bit)): CommandUtils.Run: Run: D:\Program Files\Epic Games\4.8\Engine\Binaries\DotNET\UnrealBuildTool.exe Klepto Win64 Development -clean "D:\Unreal Projects\Klepto\Klepto.uproject"  -nobuilduht -rocket -NoHotReloadFromIDE
MainFrameActions: Packaging (Windows (64-bit)): UnrealBuildTool: ERROR: Couldn't find module rules file for module 'FaceFX'.
MainFrameActions: Packaging (Windows (64-bit)): CommandUtils.Run: Run: Took 3.2881881s to run UnrealBuildTool.exe, ExitCode=5
MainFrameActions: Packaging (Windows (64-bit)): ErrorReporter.Error: ERROR: AutomationTool error: Command failed (Result:5): D:\Program Files\Epic Games\4.8\Engine\Binaries\DotNET\UnrealBuildTool.exe Klepto Win64 Development -clean "D:\Unreal Projects\Klepto\Klepto.uproject"  -nobuilduht -rocket -NoHotReloadFromIDE. See logfile for details: 'UnrealBuildTool.txt' 
MainFrameActions: Packaging (Windows (64-bit)): BuildCommand.Execute: ERROR: BUILD FAILED
MainFrameActions: Packaging (Windows (64-bit)): Program.Main: ERROR: AutomationTool terminated with exception:
MainFrameActions: Packaging (Windows (64-bit)): Program.Main: ERROR: Exception in AutomationTool: Command failed (Result:5): D:\Program Files\Epic Games\4.8\Engine\Binaries\DotNET\UnrealBuildTool.exe Klepto Win64 Development -clean "D:\Unreal Projects\Klepto\Klepto.uproject"  -nobuilduht -rocket -NoHotR
MainFrameActions: Packaging (Windows (64-bit)): eloadFromIDE. See logfile for details: 'UnrealBuildTool.txt' 
MainFrameActions: Packaging (Windows (64-bit)): Stacktrace:    at AutomationTool.CommandUtils.RunAndLog(String App, String CommandLine, String Logfile, Int32 MaxSuccessCode, String Input, ERunOptions Options, Dictionary`2 EnvVars)
MainFrameActions: Packaging (Windows (64-bit)):    at AutomationTool.CommandUtils.RunUBT(CommandEnvironment Env, String UBTExecutable, String CommandLine, String LogName, Dictionary`2 EnvVars)
MainFrameActions: Packaging (Windows (64-bit)):    at AutomationTool.UE4Build.CleanWithUBT(String ProjectName, String TargetName, UnrealTargetPlatform Platform, String Config, String UprojectPath, Boolean ForceMonolithic, Boolean ForceNonUnity, Boolean ForceDebugInfo, String InAddArgs, Boolean ForceUnity, Dictionary`2 EnvVars)
MainFrameActions: Packaging (Windows (64-bit)):    at AutomationTool.UE4Build.Build(BuildAgenda Agenda, Nullable`1 InDeleteBuildProducts, Boolean InUpdateVersionFiles, Boolean InForceNoXGE, Boolean InForceNonUnity, Boolean InForceUnity, Dictionary`2 PlatformEnvVars)
MainFrameActions: Packaging (Windows (64-bit)):    at Project.Build(BuildCommand Command, ProjectParams Params, Int32 WorkingCL)
MainFrameActions: Packaging (Windows (64-bit)):    at BuildCookRun.DoBuildCookRun(ProjectParams Params)
MainFrameActions: Packaging (Windows (64-bit)):    at BuildCommand.Execute()
MainFrameActions: Packaging (Windows (64-bit)):    at AutomationTool.Automation.Execute(List`1 CommandsToExecute, CaselessDictionary`1 Commands)
MainFrameActions: Packaging (Windows (64-bit)):    at AutomationTool.Automation.Process(String[] CommandLine)
MainFrameActions: Packaging (Windows (64-bit)):    at AutomationTool.Program.MainProc(Object Param)
MainFrameActions: Packaging (Windows (64-bit)):    at AutomationTool.InternalUtils.RunSingleInstance(MainProc Main, Object Param)
MainFrameActions: Packaging (Windows (64-bit)):    at AutomationTool.Program.Main()
MainFrameActions: Packaging (Windows (64-bit)): Program.Main: ERROR: Command failed (Result:5): D:\Program Files\Epic Games\4.8\Engine\Binaries\DotNET\UnrealBuildTool.exe Klepto Win64 Development -clean "D:\Unreal Projects\Klepto\Klepto.uproject"  -nobuilduht -rocket -NoHotReloadFromIDE. See logfile for details: 'UnrealBuildTool.txt' 
MainFrameActions: Packaging (Windows (64-bit)): ProcessManager.KillAll: Trying to kill 0 spawned processes.
MainFrameActions: Packaging (Windows (64-bit)): Program.Main: AutomationTool exiting with ExitCode=5
MainFrameActions: Packaging (Windows (64-bit)): Domain_ProcessExit
MainFrameActions: Packaging (Windows (64-bit)): ProcessManager.KillAll: Trying to kill 0 spawned processes.
MainFrameActions: Packaging (Windows (64-bit)): copying UAT log files...
MainFrameActions: Packaging (Windows (64-bit)): RunUAT.bat ERROR: AutomationTool was unable to run successfully.
MainFrameActions: Packaging (Windows (64-bit)): BUILD FAILED

--- a/Documentation/Troubleshooting.md
+++ b/Documentation/Troubleshooting.md
@@ -1,6 +1,12 @@
 Troubleshooting
 ===============
 
+#### Errors when packaging
+
+There is a known bug in UE4 where it requires all plugin Build.cs files to be present when packaging a game. This effectively means that in order to package your game you need full source to the FaceFX plugin installed.
+
+If you are using the GitHub source, this is a non-issue. However, if you are using the Epic Launcher version of the engine, it means you need to get the full source to the FaceFX plugin, including the FaceFX Runtime library, and set it up in the Epic Launcher version of your engine. Place the source code in the same location you put the pre-built binaries of the FaceFX plugin (see the [README.md](README.md) file located in this repository).
+
 #### Audio plays with no animation and there are no FaceFX errors in the log
 
 Make sure the character's **Animation Blueprint** is referenced in the **Blueprint Class**, the **Blend FaceFX Animation** Blueprint node is added to the **Animation Blueprint**, and it is connected properly.

--- a/README.md
+++ b/README.md
@@ -21,6 +21,12 @@ Supported Unreal Engine 4 versions
 
 The FaceFX UE4 plugin currently supports UE4 version 4.8.0 and up. It will not work unmodified on earlier versions of UE4.
 
+Documentation
+-------------
+
+The [FaceFX UE4 Plugin documentation](Documentation/Index.md) is located in the **Documentation** directory of this GitHub repository. Be sure to check out the [Troubleshooting](Documentation/Troubleshooting.md) section if you run into any problems.
+
+
 Installation
 ------------
 
@@ -98,11 +104,6 @@ First, make sure you are familiar with the process of cloning Unreal Engine from
 9. Load the UE4 project in Xcode. Select the **UE4Editor - Mac** for **My Mac** target in the title bar, then select the 'Product > Build' menu item.
 
 10. Run UnrealEd according to Epic's instructions.
-
-Documentation
--------------
-
-The [FaceFX UE4 Plugin documentation](Documentation/Index.md) is located in the **Documentation** directory of this GitHub repository.
 
 Contributing
 ------------

--- a/Source/FaceFX/FaceFX.Build.cs
+++ b/Source/FaceFX/FaceFX.Build.cs
@@ -29,7 +29,6 @@ public class FaceFX : ModuleRules
                 "Core",
                 "CoreUObject",
                 "Engine",
-                "TargetPlatform",
                 "FaceFXLib"
             }
         );

--- a/Source/FaceFXEditor/FaceFXEditor.Build.cs
+++ b/Source/FaceFXEditor/FaceFXEditor.Build.cs
@@ -38,7 +38,6 @@ public class FaceFXEditor : ModuleRules
                 "AssetRegistry",
                 "ContentBrowser",
                 "MainFrame",
-                "DesktopPlatform",
                 "AnimGraph",
                 "BlueprintGraph",
                 "FaceFX"

